### ISSUE TITLE
Segments: RegExp filter not works with others types

### ISSUE THREAD

**Issue by alanpy** [mautic/mautic#6639]
**Please DO NOT report security vulnerabilities here. Send them to security@mautic.com instead.**

What type of report is this:

| Q  | A
| ---| ---
| Bug report? |  X
| Feature request? | 
| Enhancement? | 

## Description:
After update from 2.12.1 to 2.14.0 all my combined segments using RegExp stops work.

When combined RegExp filter with any other filter type like "is empty" or "contains" the mautic segment gets error on processing.

```
[Symfony\Component\Debug\Exception\UndefinedMethodException]
  Attempted to call an undefined method named "regexp" of class "Doctrine\DBAL\Query\Expression\ExpressionBuilder".
```

If using only RegExp types it works.

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.14.0
| PHP version | PHP 7.0.26

### Steps to reproduce:
1. Create a new segment
2. Add RegExp Filter
3. Add another condition like email (is not empty)
4. Run cron segment update script (php -q apps/mautic/htdocs/app/console mautic:segments:update)
 
### Log errors: 

_Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable._ 

[2018-09-24 22:15:04] mautic.NOTICE: Symfony\Component\Debug\Exception\FatalThrowableError: Call to undefined method Doctrine\DBAL\Query\Expression\ExpressionBuilder::regexp() (uncaught exception) at /opt/bitnami/apps/mautic/htdocs/app/bundles/LeadBundle/Segment/Query/Filter/ForeignValueFilterQueryBuilder.php line 99 while running console command `mautic:segments:update` [] []

![regexp filter error](https://user-images.githubusercontent.com/9327067/45982574-43e2e180-c027-11e8-9c69-2a5a83c108dd.JPG)

**Comment by Enc3phale** [comment]
I think it's fixed here : https://github.com/mautic/mautic/pull/6641

**Comment by alanhartless** [comment]
Fixed in 2.14.2
